File: examples/awfy/node/sieve.mjs
export const warmup = 10, iters = 40, size = 5000, expected = 669;

export function run(size) {
  const flags = new Array(size + 1).fill(true);
  let count = 0;
  for (let i = 2; i <= size; i++) {
    if (flags[i]) {
      count++;
      for (let k = i + i; k <= size; k += i) flags[k] = false;
    }
  }
  return count;
}

File: examples/awfy/go/sieve.go
package main

func sieveRun(size int) int {
	flags := make([]bool, size+1)
	for i := range flags {
		flags[i] = true
	}
	count := 0
	for i := 2; i <= size; i++ {
		if flags[i] {
			count++
			for k := i + i; k <= size; k += i {
				flags[k] = false
			}
		}
	}
	return count
}

var sieveBench = Bench{Name: "sieve", Warmup: 10, Iters: 40, Size: 5000, Expected: 669, Run: sieveRun}
